src/db.rs

```diff
@@ -98,6 +98,38 @@ impl DbInner {
         Ok(None)
     }
 
+    async fn fence_writers(&self, manifest: &mut FenceableManifest) -> Result<(), SlateDBError> {
+        let wal_id_last_compacted = self.state.read().state().core.last_compacted_wal_sst_id;
+        let max_wal_id = self
+            .table_store
+            .get_wal_sst_list(wal_id_last_compacted)
+            .await?
+            .into_iter()
+            .max()
+            .unwrap_or(0);
+        let mut empty_wal_id = max_wal_id + 1;
+
+        loop {
+            let empty_wal = WritableKVTable::new();
+            match self
+                .flush_imm_table(&SsTableId::Wal(empty_wal_id), empty_wal.table().clone())
+                .await
+            {
+                Ok(_) => {
+                    return Ok(());
+                }
+                Err(SlateDBError::Fenced) => {
+                    let updated_state = manifest.refresh().await?;
```

Comment:
I'm confused - why do we need to do this here? If we're fenced, we need to check the epoch that `empty_wal_id` was written with, and then either exit with an error because that epoch is higher than our epoch, or move on to the next wal id. There is no way the manifest could be updated in a way that it makes sense to consume it at this point. It cannot have been updated by an older writer because we already updated the epoch in the manifest.

Reply:
Fair enough. I agree it's not a huge deal to read the manifest again. We should have the wal segment's epoch handy at this point right (when that's implemented), since it's returned when reading the wal object? So then we wouldn't need the extra lookup, and the code becomes simpler. So I still prefer to just use that if it's there. But we can change it when we have the object metadata stuff implemented. Can you file a ticket?

Reply:
Opened
#153 and #154

Reply:
> We should have the wal segment's epoch handy at this point right (when that's implemented), since it's returned when reading the wal object? So then we wouldn't need the extra lookup, and the code becomes simpler

Agreed.